test/e2e-olm/elasticsearch_test.go

```diff
@@ -1,159 +1,240 @@
 package e2e
 
 import (
+	"context"
 	"fmt"
 	"testing"
-	"time"
 
+	loggingv1 "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
 	"github.com/openshift/elasticsearch-operator/test/utils"
-	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
-	"k8s.io/apimachinery/pkg/api/resource"
-	"k8s.io/apimachinery/pkg/types"
 
-	goctx "context"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
 
-	elasticsearch "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
-	framework "github.com/operator-framework/operator-sdk/pkg/test"
-	v1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func TestElasticsearch(t *testing.T) {
+func TestElasticsearchCluster(t *testing.T) {
 	registerSchemes(t)
-	t.Run("elasticsearch-group", func(t *testing.T) {
-		t.Run("Cluster", ElasticsearchCluster)
-	})
+	t.Run("Single node", singleNodeTest)
+	t.Run("Multiple nodes", multipleNodesTest)
+	t.Run("Multiple nodes with a single non-data node", multipleNodesWithNonDataNodeTest)
+	t.Run("Full cluster redeploy", fullClusterRedeployTest)
+	t.Run("Rolling restart", rollingRestartTest)
+	t.Run("Invalid master count", invalidMasterCountTest)
 }
 
-func elasticsearchFullClusterTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+func singleNodeTest(t *testing.T) {
+	f := test.Global
+
+	ctx := test.NewTestCtx(t)
 	namespace, err := ctx.GetNamespace()
 	if err != nil {
-		return fmt.Errorf("Could not get namespace: %v", err)
+		t.Fatal(err)
+	}
+	t.Logf("Found namespace: %v", namespace)
+
+	if err = createElasticsearchSecret(t, f, ctx); err != nil {
```

Comment:
this shouldn't change for each test, would it make sense to break this out into some sort of "setup" step? then for a cert full cluster restart we could regenerate as part of that test...

Reply:
All calls to `createElasticsearchSecret` register a cleanup routing for the secret. In addition each testcase runs `ctx.Cleanup()` at the end. Thus creating the secret is part of each test case. Since this only creates a secret resource w/o re-generating the certs it is ok to re-create on each test. This ensures that each test is independent.